scalableminds/webknossos
# Strict permission checks during all steps of dataset upload

Tightened and cleaned up security for dataset uploads
 - for every chunk, and finishUpload, the token is validated against the upload id (don’t worry, there is a cache)
 - the initial teams timeout check was removed (since it has already previously been moved to the reserveUpload request which can be sent only once per dataset)
 - administrateDataSources access checks are now organization-specific for better isolation
 - removed unused (and broken) datasource list route

### Steps to test:
- upload a dataset as admin or dataset manager or team manager, should go smoothly, including setting teams
- try to forge requests screwing with anyone else’s uploads

### Issues:
- fixes #5787

------
